src/datasets/load.py

```diff
@@ -733,20 +733,25 @@ def __init__(
         data_files: Optional[Union[str, List, Dict]] = None,
         download_mode: Optional[DownloadMode] = None,
     ):
-        if data_files is None and data_dir is not None:
-            data_files = os.path.join(data_dir, "**")
 
         self.path = path
         self.name = Path(path).stem
         self.data_files = data_files
+        self.data_dir = data_dir if data_dir else None
         self.download_mode = download_mode
 
     def get_module(self) -> DatasetModule:
         patterns = (
-            sanitize_patterns(self.data_files) if self.data_files is not None else get_patterns_locally(self.path)
+            sanitize_patterns(self.data_files)
+            if self.data_files is not None
+            else get_patterns_locally(self.data_dir)
```

Comment:
If `self.data_dir` is a relative path, I think it should be relative to the dataset repository's root `self.path`, for consistency with HubDatasetModuleFactoryWithoutScript

Reply:
[6fb34ff](https://github.com/huggingface/datasets/pull/4144/commits/6fb34ff72fd05046897fd6a17399fd4d7709863e)

[0fa3cd8](https://github.com/huggingface/datasets/pull/4144/commits/0fa3cd8a4d7a6d8947479e4f4196cdadc2a27aa5)